Pseudocode/Queue/First Non Repeating in Stream/SourceCode.tex

```diff
@@ -0,0 +1,59 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Remove the 'end' at the end of the algorithm
+\usepackage[noend]{algpseudocode}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+    \begin{algorithm}
+    
+      \caption{Find the first non-repeating character when each character of the stream arrives}
+      
+      \begin{algorithmic}[1]
+        \Ensure {All characters of string $str$ are lowercase alphabets}
```

Comment:
It's better to handle a general case.

My suggestion is to use a **HashMap** called **frequency** to store the frequency instead of the count array.